sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -294,11 +304,22 @@ private void decodeDictionaryIds(
         break;
       case INT96:
         if (column.dataType() == DataTypes.TimestampType) {
-          for (int i = rowId; i < rowId + num; ++i) {
-            // TODO: Convert dictionary of Binaries to dictionary of Longs
-            if (!column.isNullAt(i)) {
-              Binary v = dictionary.decodeToBinary(dictionaryIds.getDictId(i));
-              column.putLong(i, ParquetRowConverter.binaryToSQLTimestamp(v));
+          if (!shouldConvertTimestamps()) {
+            for (int i = rowId; i < rowId + num; ++i) {
+              if (!column.isNullAt(i)) {
+                Binary v = dictionary.decodeToBinary(dictionaryIds.getDictId(i));
+                column.putLong(i, ParquetRowConverter.binaryToSQLTimestamp(v));
+              }
+            }
+          } else {
+            for (int i = rowId; i < rowId + num; ++i) {
+              // TODO: Convert dictionary of Binaries to dictionary of Longs
```

Comment:
I think this comment belongs to https://github.com/apache/spark/pull/19769/files#diff-7bdf5fd0ce0b1ccbf4ecf083611976e6R308 ?

Reply:
Yep, I've removed it because I think the idea needs a bit more discussion re: tradeoffs. I filed [SPARK-22736](https://issues.apache.org/jira/browse/SPARK-22736).